ratis-metrics/src/main/java/org/apache/ratis/metrics/impl/MetricRegistriesImpl.java

```diff
@@ -61,11 +61,6 @@ public MetricRegistriesImpl(MetricRegistryFactory factory) {
   @Override
   public RatisMetricRegistry create(MetricRegistryInfo info) {
     return registries.put(info, () -> {
-      if (reporterRegistrations.isEmpty()) {
-        LOG.warn(
-            "First MetricRegistry has been created without registering reporters. You may need to call" +
-                " MetricRegistries.global().addReporterRegistration(...) before.");
-      }
```

Comment:
Doesn't the warning point to a real problem?  Are client message metrics being reported?

Reply:
So without reported being added, I believe there would not be any metrics to report. But then, it should be fixed from the calling function. Currently, if we run a load on the system, we keep getting these warning messages from ratis clients. It does not serve the purpose to spam the client logs with this warning message, as it needs to fixed on server side.

Reply:
bq. Currently, if we run a load on the system, we keep getting these warning messages from ratis clients. It does not serve the purpose to spam the client logs with this warning message, as it needs to fixed on server side.

I am fine with removing the warning, but can you please give more information how can the problem be reproduced? As far as I remember I also saw it earlier, but today I just tested with `ozone sh` and `ozonen freon` and couldn't reproduce.

Do we have any racecondition somewhere between adding reporters + registering the first metric? Or is the metric registration missing from somewhere?

Reply:
I don't think it can be fixed on the server side.  Reporters are registered with `MetricRegistries.global()`, which is a singleton limited to the Java process (i.e. here the client).